halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/config/v1/HalconfigParser.java

```diff
@@ -166,30 +136,15 @@ public void saveConfig() {
       );
     }
 
-    Writer writer = null;
     try {
-      writer = new BufferedWriter(
-          new OutputStreamWriter(new FileOutputStream(halconfigPath), "utf-8"));
-      writer.write(yamlParser.dump(objectMapper.convertValue(halconfig, Map.class)));
-    } catch (Exception e) {
+      fileWriter.write(yamlParser.dump(objectMapper.convertValue(halconfig, Map.class)), halconfigPath);
```

Comment:
An atomic disk write is a mv(2) where you are just swapping inodes but the content is already there. I dont think you can assume write(2) is atomic. You dont need the builder per say. I think you need a temporary file as an intermediary.